### PR TITLE
MAINT, BUG: Final 1.14 formatting fixes

### PR DESCRIPTION
**Pull request by eric-wieser**
Follows on from gh-10176, and does the missed release notes edit for that and some other PRs.

Also tries to group the changes together in the release notes.

### CODE REVIEW

File: doc/release/1.14.0-notes.rst [hunk 1]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
+  * The "dtype" part of ndarray reprs will now be printed on the next line
+    if there isn't space on the last line of array output.
+  * The ``linewidth`` format option is now always respected.
+    The `repr` or `str` of an array will never exceed this, unless a single
+    element is too wide.
+  * All but the last line of array strings will contain the same number of
+    elements.
+  * The last line of an array string will never have more elements than earlier
+    lines.
+* To summarization, the use of ``...`` to shorten long arrays:
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: summarization <- summarize.

File: doc/release/1.14.0-notes.rst [hunk 2]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
-* Floating-point arrays and scalars use a new algorithm for decimal
-  representations, giving the shortest unique representation. This will
-  usually shorten ``float16`` fractional output, and sometimes ``float32`` and
-  ``float128`` output. ``float64`` should be unaffected.  See the new
-  ``floatmode`` option to ``np.set_printoptions``.
-* Float arrays printed in scientific notation no longer use fixed-precision,
-  and now instead show the shortest unique representation.
-* The ``str`` of floating-point scalars is no longer truncated in python2.
-* Non-finite complex scalars print like ``nanj`` instead of ``nan*j``.
+* To floating-point types:
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: Indenting the list results in a line down the left hand side when rendered for documentation , which is why it wasn't indented before. The sublists need to be separated by blank lines. If you view the file you can see some of these problems.

File: doc/release/1.14.0-notes.rst [hunk 3]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
+    in the sign position. See the new ``sign`` option to ``np.set_printoptions``.
+  * Floating-point arrays and scalars use a new algorithm for decimal
+    representations, giving the shortest unique representation. This will
+    usually shorten ``float16`` fractional output, and sometimes ``float32`` and
+    ``float128`` output. ``float64`` should be unaffected.  See the new
+    ``floatmode`` option to ``np.set_printoptions``.
+  * Float arrays printed in scientific notation no longer use fixed-precision,
+    and now instead show the shortest unique representation.
+  * The ``str`` of floating-point scalars is no longer truncated in python2.
+* To other data types:
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: "For" other data types.

File: doc/release/1.14.0-notes.rst (hunk 2 above)
Comment (charris) [reply doc/release/1.14.0-notes.rst]: Generally, "to" should be replaced by "for".

File: doc/release/1.14.0-notes.rst (hunk 1 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: I want the noun form here, since the other list items are nouns.

**Review by mhvk — APPROVED** [review]
Looks all OK to me.

File: doc/release/1.14.0-notes.rst (hunk 2 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: Thanks for reminding me that github can render RST files.

File: doc/release/1.14.0-notes.rst (hunk 1 above)
Comment (mhvk) [reply doc/release/1.14.0-notes.rst]: Maybe "For summarization"

File: doc/release/1.14.0-notes.rst (hunk 3 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: I used "to" for consistency with the other headings

File: doc/release/1.14.0-notes.rst [hunk 4]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
+    The `repr` or `str` of an array will never exceed this, unless a single
+    element is too wide.
+  * All but the last line of array strings will contain the same number of
+    elements.
+  * The last line of an array string will never have more elements than earlier
+    lines.
+* To summarization, the use of ``...`` to shorten long arrays:
+  * A trailing comma is no longer inserted for ``str``.
+    Previously, ``str(np.arange(1001))`` gave
+    ``'[   0    1    2 ...,  998  999 1000]'``, which has an extra comma.
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: which <- that.

File: doc/release/1.14.0-notes.rst [hunk 5]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
+  * For arrays of 2-d and beyond, the ``...`` printed on its own line to
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: "is printed"? Missing period after "line".

File: doc/release/1.14.0-notes.rst [hunk 6]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
+    summarize all but the last axis now has trailing newlines added to match
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: Comma after summarize.

File: doc/release/1.14.0-notes.rst [hunk 7]
```diff
@@ -253,37 +253,50 @@ enabling the new 1.13 "legacy" printing mode. This is enabled by calling
...
+    the leading newlines, and the trailing space character removed.
```
Comment (charris) [inline doc/release/1.14.0-notes.rst]: "is removed"

**Comment by charris** [comment]
The release notes need fixing up.

File: doc/release/1.14.0-notes.rst (hunk 7 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: Not convinced - "the ... on its own line has ... the trailing space character removed"

**Comment by eric-wieser** [comment]
Formatting and To/for fixed. Can I get you to make the final fixups?

File: doc/release/1.14.0-notes.rst (hunk 4 above)
Comment (charris) [reply doc/release/1.14.0-notes.rst]: NVM, I think "which" is correct here.

File: doc/release/1.14.0-notes.rst (hunk 6 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: Also disagree here.

File: doc/release/1.14.0-notes.rst (hunk 1 above)
Comment (charris) [reply doc/release/1.14.0-notes.rst]: "to noun" doesn't read right, "to nounize" does, even though "nounize" isn't a word ;)

File: doc/release/1.14.0-notes.rst (hunk 5 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: Nope, this is deliberate

File: doc/release/1.14.0-notes.rst (hunk 1 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: This was intended to be read as _"the changes are:"_ .. _"to summarization"_, which reads just fine. But perhaps that kind of continuation from list heading to item is confusing. I've replaced all of the _"to"_s with _"for"_s

File: doc/release/1.14.0-notes.rst (hunk 7 above)
Comment (ahaldane) [reply doc/release/1.14.0-notes.rst]: The sentence is hard to read, I would just rework it somehow . Maybe

> For arrays of 2-d and beyond, when ``...`` is printed on its own line to summarize any but the last axis,  trailing newlines are now added to match the leading newlines, and the trailing space character is removed.

**Comment by ahaldane** [comment]
LGTM except that sentence.

File: doc/release/1.14.0-notes.rst (hunk 7 above)
Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: To me, that seems to imply the newlines are added to the array, not to the `...`

Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: Parenthesize "(to summarize all but the last axis)"?

Comment (ahaldane) [reply doc/release/1.14.0-notes.rst]: > For arrays of 2-d and beyond, when `...` is printed on its own line in order to summarize any but the last axis, newlines are now appended to that line to match its leading newlines, and a trailing space character is removed.

?

Comment (eric-wieser) [reply doc/release/1.14.0-notes.rst]: Sounds good. Can you fix up?

**Comment by charris** [comment]
Thanks Eric.

**Comment by jakirkham** [comment]
So when is 1.14.0 scheduled to be released? Are there any RCs planned? Is there an issue I should be following for this sort of stuff?

**Comment by charris** [comment]
@jakirkham I will try to get an rc1 release out tomorrow. When the final comes out will depend on how things go from there, but I wouldn't expect it to be out until Jan.

**Comment by jakirkham** [comment]
Ok, well please ping me when the RC comes out. I'd like to put the dev package on `conda-forge` so that you can get some more feedback.

**Pull request merged**